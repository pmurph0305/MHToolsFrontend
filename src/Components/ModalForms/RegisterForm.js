import React from "react";
import PropTypes from "prop-types";
import "./SignInForm.scss";
/** Register form - returns a component that renders a registration form
 * @param  {string} formError error text displayed on the form
 * @callback onSubmitForm called when the user clicks the submit button
 * @callback onToggleModal called when the user clicks the close (X) button on the modal.
 */
const RegisterForm = ({ onSubmitForm, onToggleModal, formError }) => {
  return (
    <div className="form-container">
      <form
        className="form-modal"
        action=""
        method="POST"
        onSubmit={onSubmitForm}
      >
        <div className="modal-close" onClick={onToggleModal}>
          &times;
        </div>
        <div className="form-flex-container">
          <h1 className="form-title">Register</h1>
          <div className="form-entries-container">
            <div className="form-input-container">
              <label className="form-label" htmlFor="email">
                Email:
              </label>
              <input
                autoComplete="email"
                className="form-input"
                type="email"
                name="email"
                id="email"
                required
              />
            </div>
            <div className="form-input-container">
              <label className="form-label" htmlFor="username">
                Username:
              </label>
              <input
                className="form-input"
                type="text"
                name=""
                id=""
                required
              />
            </div>
            <div className="form-input-container">
              <label className="form-label" htmlFor="password">
                Password:
              </label>
              <input
                autoComplete="new-password"
                className="form-input"
                type="password"
                name="password"
                id="password"
                required
              />
            </div>
            <div className="form-input-container">
              <label className="form-label" htmlFor="password2">
                Re-enter Password:
              </label>
              <input
                autoComplete="new-password"
                className="form-input"
                type="password"
                name="password2"
                id="password2"
                required
              />
            </div>
            <input
              type="hidden"
              id="hiddenFormEl"
              name="hidden"
              value="hiddenValue"
            />
            {formError}
            <input className="form-submit" type="submit" value="Register" />
          </div>
        </div>
      </form>
    </div>
  );
};

RegisterForm.propTypes = {
  onSubmitForm: PropTypes.func.isRequired,
  onToggleModal: PropTypes.func.isRequired,
  formError: PropTypes.string
};

export default RegisterForm;
