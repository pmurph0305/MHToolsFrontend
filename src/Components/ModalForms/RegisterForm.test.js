import { shallow } from "enzyme";
import React from "react";
import RegisterForm from "./RegisterForm";

describe("RegisterForm tests", () => {
  let mockOnSubmit = jest.fn(val => val);
  let mockOnToggleModal = jest.fn(val => val);
  let mockProps = {
    onSubmitForm: mockOnSubmit,
    onToggleModal: mockOnToggleModal
  };
  const wrapper = shallow(<RegisterForm {...mockProps} />);

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onsubmit", () => {
    wrapper.find("form").simulate("submit");
    expect(mockOnSubmit).toHaveBeenCalledTimes(1);
  });

  it("calls toggle properly", () => {
    wrapper.find('[className="modal-close"]').simulate("click");
    expect(mockOnToggleModal).toHaveBeenCalledTimes(1);
  });
});
