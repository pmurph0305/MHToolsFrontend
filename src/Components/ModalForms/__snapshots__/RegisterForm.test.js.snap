// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm tests matches snapshot 1`] = `ShallowWrapper {}`;
